WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -158,33 +224,40 @@ private class WeeklyRoundupDataProvider {
             return
         }
 
-        filterWeeklyRoundupEnabledSites(administeredSites, result: result)
+        filterWeeklyRoundupEnabledSites(administeredSites, in: context, result: result)
     }
 
     /// Filters the sites that have the Weekly Roundup notification setting enabled.
     ///
-    private func filterWeeklyRoundupEnabledSites(_ sites: [Blog], result: @escaping (Result<[Blog], Error>) -> Void) {
+    private func filterWeeklyRoundupEnabledSites(_ sites: [Blog], in context: NSManagedObjectContext, result: @escaping (Result<[Blog], Error>) -> Void) {
         let noteService = NotificationSettingsService(coreDataStack: coreDataStack)
 
         noteService.getAllSettings { settings in
-            let weeklyRoundupEnabledSites = sites.filter { site in
-                guard let siteSettings = settings.first(where: { $0.blog == site }),
-                      let pushNotificationsStream = siteSettings.streams.first(where: { $0.kind == .Device }),
-                      let sitePreferences = pushNotificationsStream.preferences else {
-                    return false
-                }
-
-                return sitePreferences["weekly_roundup"] ?? true
+            context.perform {
+                let weeklyRoundupEnabledSites = Self.weeklyRoundupEnabledSites(settings: settings, sites: sites, in: context)
+                result(.success(weeklyRoundupEnabledSites))
             }
-
-            result(.success(weeklyRoundupEnabledSites))
         } failure: { (error: NSError?) in
             let error = DataRequestError.filterWeeklyRoundupEnabledSitesError(error)
             result(.failure(error))
```

Comment:
The `result` closure here should be called from the `context` queue too?

Reply:
See https://github.com/wordpress-mobile/WordPress-iOS/pull/21076#discussion_r1267095822

Reply:
Resolving this conversation as this change is no longer needed. See this [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/21076#discussion_r1267526635).